topic-controller/src/main/java/io/strimzi/controller/topic/TopicMetadataHandler.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.strimzi.controller.topic;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Represents an handler for getting Kafka topic metadata with built-in retry
```

Comment:
pedantic but "a handler" rather than "an handler" (using 'an' with a word beginning with h is becoming pretty old fashioned in British English)

Reply:
"with built-in retry mechanism", strictly it's not really "built-in" (this isn't like the template method pattern; subclasses won't magically get retry behaviour), rather you've added a a helper method for subclasses to use when they want to retry.

Reply:
About "a handler" you already said me about that ... I'm not a good english student :)

Reply:
@ppatierno it's my secret plan to make you a better one :-)